src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -33,17 +34,19 @@ class Router extends BaseRouter implements WarmableInterface, ServiceSubscriberI
     private $collectedParameters = array();
 
     /**
-     * @param ContainerInterface $container A ContainerInterface instance
-     * @param mixed              $resource  The main resource to load
-     * @param array              $options   An array of options
-     * @param RequestContext     $context   The context
+     * @param ContainerInterface   $container A ContainerInterface instance
+     * @param mixed                $resource  The main resource to load
+     * @param array                $options   An array of options
+     * @param RequestContext       $context   The context
+     * @param LoggerInterface|null $logger
      */
-    public function __construct(ContainerInterface $container, $resource, array $options = array(), RequestContext $context = null)
+    public function __construct(ContainerInterface $container, $resource, array $options = array(), RequestContext $context = null, LoggerInterface $logger = null)
     {
         $this->container = $container;
 
         $this->resource = $resource;
         $this->context = $context ?: new RequestContext();
+        $this->logger = $logger;
```

Comment:
Just wondering: Why don't we call the parent class' constructor?

Reply:
The `$loader` argument is mandatory in parent constructor, but in this implementation we're lazy-loading it.

Reply:
Thanks, I missed that.

Reply:
We should probably change this in 4.1 to have a lazy-loading implementation of the LoaderInterface (using composition over the actual loader which would get loaded when needed), to avoid having a special constructor.